Makefile

```diff
@@ -0,0 +1,18 @@
+CC = gcc
+CFLAGS = -g -Wall -Werror -fPIC
+DEPS = resource.h res_impl.h resmem.h resnet.h
+OBJ = resource.o resmem.o resnet.o
+LIB = libresource.so
+TEST = test
+RM = rm -rf
+CP = cp
+
+%.o: %.c $(DEPS)
+	$(CC) -c -o $@ $< $(CFLAGS)
+
+all: $(OBJ)
+	$(CC) -shared -o $(LIB) $^ $(CFLAGS)
+
+.PHONY : clean
+clean:
+	$(RM) $(LIB) $(OBJ) $(TEST)
```

Comment:
I guess that'll do for now but we'll need a proper build system such as autotools or meson. Whatever works.

Reply:
I agree. I have started workng on such build system using autotools and can push it with next set of resource ids after this goes in.